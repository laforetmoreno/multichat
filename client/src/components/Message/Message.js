import React from 'react';
import { object } from 'prop-types';

import styles from './Message.module.scss';

const Message = ({ message, user }) => (
  <li className={styles.item} key={message.value}>
    <div className={`${styles.messageContainer} ${message.author === user.nickName && styles.myMessage}`}>
      <strong className={styles.author}>{message.author}</strong>
      <span className={styles.value}>{message.value}</span>
    </div>
  </li>
);

Message.propTypes = {
  message: object,
  user: object,
};

export default Message;
