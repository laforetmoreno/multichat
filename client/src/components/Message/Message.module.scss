@import '../../styles/variables';

.item {
  .messageContainer {
    padding: 5px 20px;
    border-radius: 4px;
    background: $message-color;
    max-width: 170px;
    float: left;
    text-align: left;
    margin: 5px 0;
  }

  .author,
  .value {
    display: block;
  }
}

.myMessage {
  float: right !important;
  text-align: right !important;
  background: $my-message-color !important;
}
