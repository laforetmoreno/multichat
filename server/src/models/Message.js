const mongoose = require('../db');

const MessageSchema = new mongoose.Schema({
  value: {
    type: String,
    require: true,
  },
  author: {
    type: String,
    require: true,
  },
  room: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Room',
    require: true,
  },
  createAt: {
    type: Date,
    default: Date.now,
  },
});

const User = mongoose.model('Message', MessageSchema);

module.exports = User;
