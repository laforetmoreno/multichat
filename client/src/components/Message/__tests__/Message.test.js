import React from 'react';
import Message from '../Message';
import renderer from 'react-test-renderer';

describe('<Message />', () => {
  it('renders correctly', () => {
    const messageFake = {
      value: 'Hello',
      author: 'Joe'
    }

    const userFake = {
      nickName: 'joe joe'
    }

    const tree = renderer
      .create(<Message message={messageFake} user={userFake} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});