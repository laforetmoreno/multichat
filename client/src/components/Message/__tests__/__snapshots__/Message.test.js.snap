// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> renders correctly 1`] = `
<li
  className="item"
>
  <div
    className="messageContainer false"
  >
    <strong
      className="author"
    >
      Joe
    </strong>
    <span
      className="value"
    >
      Hello
    </span>
  </div>
</li>
`;
